src/CloudNative.CloudEvents.NewtonsoftJson/JsonEventFormatter.cs

```diff
@@ -516,11 +515,11 @@ public override void DecodeBinaryModeEventData(byte[] body, CloudEvent cloudEven
             }
             else if (contentType.MediaType.StartsWith("text/") == true)
             {
-                cloudEvent.Data = encoding.GetString(body);
+                cloudEvent.Data = BinaryDataUtilities.GetString(body, encoding);
             }
             else
             {
-                cloudEvent.Data = body;
+                cloudEvent.Data = body.ToArray();
```

Comment:
Should we just leave it as ReadOnlyMemory?

Reply:
We *could* do, but there are some issues. My understanding is that a `ReadOnlyMemory` may be "owned" and then disposed, which would be bad news for anything still using it. (Or it could end up being a buffer that's reused later and modified, for example.) Arguably returning the original byte array was a mistake in the previous implementation.

It also makes the choice of `ReadOnlyMemory` (which isn't currently as user-friendly as `byte[]`) more visible to end-users. I'm okay with using it in `CloudEventFormatter` which is mostly used by other CloudEvents code, but I think end-user code is more likely to find byte arrays easy to use.

Feel free to argue against me though :)